### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/ava#160]
We've worked hard on making the documentation good, but we're biased as we made this thing and don't see it the way a first time users see it.

Documentation improvements of any kind are much appreciated. Anything to make things more explicit, more examples, fix grammar, anything.

**Comment by jamestalmage** [comment]
Are you open to github-pages documentation, or do you want to keep it all in the README?

http://documentation.js.org/ looks interesting. (particularly [this demo](https://www.mapbox.com/mapbox-gl-js/api/))

Also, you probably want to reserve `ava` at https://dns.js.org/ (that would get you https://ava.js.org)

**Comment by sindresorhus** [comment]
> Are you open to github-pages documentation, or do you want to keep it all in the README?

I would prefer to keep it in the readme for now. I plan on having a website for AVA, but not until after 1.0.0. I want to focus on AVA itself for now. Marketing can be prioritized later.

> http://documentation.js.org/ looks interesting. (particularly this demo)

Indeed. There's also https://readme.io/ (free for open-source).

> Also, you probably want to reserve ava at https://dns.js.org/ (that would get you https://ava.js.org)

I already have https://ava.li.

**Comment by jamestalmage** [comment]
Wow, documentation.js produces really nice output.

Even the [base output](http://documentation.js.org/demo/demo-documentation/) is pretty slick. Each method has a link to the source on GitHub (that works really well). Base types automatically link to `developer.mozilla.org`

**Comment by jamestalmage** [comment]
Another plus for generation API docs using some JSDoc based system is that it improves typing hints on Webstorm significantly (I am assuming other IDE's as well).

**Comment by sindresorhus** [comment]
Agreed, documentation.js looks really good.

Not a big fan of JSDoc. If I wanted type-hints I would go with [Facebook Flow](https://github.com/facebook/flow).

**Comment by sindresorhus** [comment]
@jamestalmage I opened a separate issue for a future website: https://github.com/sindresorhus/ava/issues/162

**Comment by jamestalmage** [comment]
The documentation should cover how to get AVA working with production code.

There is a lot of confusion #111 

It should incorporate the code from [this comment](https://github.com/sindresorhus/ava/issues/111#issuecomment-155334846).

It gets the point across pretty quick

**Comment by sotojuan** [comment]
I said this on Gitter but putting it out here for further discussion:

In the async function support section, I think a short explanation of what `async`/`await` functions could be helpful for newbies. We could add something about how they make asynchronous code look synchronous. Perhaps add one more "real world" example to the examples there already? I just want people to find out about `async`/`await` and not just gloss over the section!

**Comment by sindresorhus** [comment]
@sotojuan I'm all for that, but would prefer to keep it succinct and rather link to a good blog post or two with more comprehensive explanation.

**Comment by sindresorhus** [comment]
We're now also looking for people interested in translating the docs: https://github.com/sindresorhus/ava-docs
